app/api/data_layers/ChargesLayer.py

```diff
@@ -1,23 +1,45 @@
 from flask_rest_jsonapi.data_layers.base import BaseDataLayer
+from flask_rest_jsonapi.exceptions import ObjectNotFound
 
-from app.api.helpers.exceptions import UnprocessableEntity
+from app.api.helpers.exceptions import UnprocessableEntity, ConflictException
 from app.api.helpers.ticketing import TicketingManager
 from app.models.order import Order
 
 
 class ChargesLayer(BaseDataLayer):
 
     def create_object(self, data, view_kwargs):
+        """
+        create_object method for the Charges layer
+        charge the user using paypal or stripe
+        :param data:
+        :param view_kwargs:
+        :return:
+        """
         order = Order.query.filter_by(id=view_kwargs['id']).first()
+        if not order:
+            raise ObjectNotFound({'parameter': 'id'},
+                                 "Order with id: {} not found".format(view_kwargs['id']))
+        elif order.status == 'cancelled' or order.status == 'expired':
+            raise ConflictException({'parameter': 'id'},
+                                    "You cannot charge payments on a cancelled or expired order")
+        elif (not order.amount) or order.amount == 0:
+            raise ConflictException({'parameter': 'id'},
+                                    "You cannot charge payments on a free order")
```

Comment:
I know that internally the object is called order, but if you read it as an end user, it'd make more sense if you say either, free ticket or free event.

Reply:
An order may contain more than one ticket. For e.g. 4 tickets of type 1 and 3 tickets of type 2. Also, on the Frontend we are using `Order` name only for the end user, so I don't see any reason why it would confuse the user.